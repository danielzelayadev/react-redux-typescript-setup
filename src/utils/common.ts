import * as R from 'ramda';

export const getPropCurried = R.curryN(2, R.prop);
export const noop = () => {
  return;
};
