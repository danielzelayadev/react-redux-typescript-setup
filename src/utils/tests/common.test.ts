import * as utils from '../common';

describe('Common Utils', () => {
  describe('getPropCurried', () => {
    it('should return a curried R.prop function', () => {
      const getX = utils.getPropCurried('x');
      const expected = 10;
      const actual = getX({ x: expected });

      expect(actual).toBe(expected);
    });
  });
  describe('noop', () => {
    it('should do nothing', () => {
      expect({ noop: utils.noop.toString() }).toMatchSnapshot();
    });
  });
});
