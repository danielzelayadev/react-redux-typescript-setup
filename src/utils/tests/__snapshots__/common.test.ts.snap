// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Utils noop should do nothing 1`] = `
Object {
  "noop": "function () {
    return;
}",
}
`;
